### PR TITLE
feat: add Rust code in queue.md and deque.md

### PR DESCRIPTION
**Pull request by hts0000**
If this PR is related to coding or code translation, please fill out the checklist and paste the console outputs to the PR.

- [x] I've tested the code and ensured the outputs are the same as the outputs of reference codes.
- [x] I've checked the codes (formatting, comments, indentation, file header, etc) carefully.
- [x] The code does not rely on a particular environment or IDE and can be executed on a standard system (Win, macOS, Ubuntu).

### CODE REVIEW

File: docs/chapter_stack_and_queue/deque.md [hunk 1]
```diff
@@ -301,7 +301,33 @@
 === "Rust"
 
     ```rust title="deque.rs"
+    // 初始化双向队列
```
Comment (night-cruise) [inline docs/chapter_stack_and_queue/deque.md:304]: Use the /**/ style comment: `/* 初始化双向队列 */`

File: docs/chapter_stack_and_queue/deque.md [hunk 2]
```diff
@@ -301,7 +301,33 @@
...
+    let mut deque: VecDeque<u32> = VecDeque::new();
 
+    // 元素入队
```
Comment (night-cruise) [inline docs/chapter_stack_and_queue/deque.md:307]: /* 元素入队 */

File: docs/chapter_stack_and_queue/deque.md [hunk 3]
```diff
@@ -301,7 +301,33 @@
...
+    deque.push_back(2);  // 添加至队尾
+    deque.push_back(5);
+    deque.push_back(4);
+    deque.push_front(3); // 添加至队首
+    deque.push_front(1);
+
+    // 访问元素
```
Comment (night-cruise) [inline docs/chapter_stack_and_queue/deque.md:314]: /* 访问元素 */

File: docs/chapter_stack_and_queue/deque.md [hunk 4]
```diff
@@ -301,7 +301,33 @@
...
+    if let Some(front) = deque.front() { // 队首元素
+    }
+    if let Some(rear) = deque.back() {   // 队尾元素
+    }
+
+    // 元素出队
```
Comment (night-cruise) [inline docs/chapter_stack_and_queue/deque.md:320]: /* 元素出队 */

File: docs/chapter_stack_and_queue/deque.md [hunk 5]
```diff
@@ -301,7 +301,33 @@
...
+    if let Some(pop_front) = deque.pop_front() { // 队首元素出队
+    }
+    if let Some(pop_rear) = deque.pop_back() {   // 队尾元素出队
+    }
+
+    // 获取双向队列的长度
```
Comment (night-cruise) [inline docs/chapter_stack_and_queue/deque.md:326]: /* 获取双向队列的长度 */

File: docs/chapter_stack_and_queue/deque.md [hunk 6]
```diff
@@ -301,7 +301,33 @@
...
+    let size = deque.len();
+
+    // 判断双向队列是否为空
```
Comment (night-cruise) [inline docs/chapter_stack_and_queue/deque.md:329]: /* 判断双向队列是否为空 */

File: docs/chapter_stack_and_queue/queue.md [hunk 7]
```diff
@@ -268,7 +268,30 @@
 === "Rust"
 
     ```rust title="queue.rs"
-
+    // 初始化双向队列
```
Comment (night-cruise) [inline docs/chapter_stack_and_queue/queue.md:271]: /* 初始化队列 */

File: docs/chapter_stack_and_queue/queue.md [hunk 8]
```diff
@@ -268,7 +268,30 @@
...
+    // 在 Rust 中使用双向队列作为普通队列来使用
+    let mut deque: VecDeque<u32> = VecDeque::new();
+
+    // 元素入队
```
Comment (night-cruise) [inline docs/chapter_stack_and_queue/queue.md:275]: /* 元素入队 */

File: docs/chapter_stack_and_queue/queue.md [hunk 9]
```diff
@@ -268,7 +268,30 @@
...
+    deque.push_back(1);
+    deque.push_back(3);
+    deque.push_back(2);
+    deque.push_back(5);
+    deque.push_back(4);
+
+    // 访问队首元素
```
Comment (night-cruise) [inline docs/chapter_stack_and_queue/queue.md:282]: /* 访问队首元素 */

File: docs/chapter_stack_and_queue/queue.md [hunk 10]
```diff
@@ -268,7 +268,30 @@
...
+    if let Some(front) = deque.front() {
+    }
+
+    // 元素出队
```
Comment (night-cruise) [inline docs/chapter_stack_and_queue/queue.md:286]: /* 元素出队 */

File: docs/chapter_stack_and_queue/queue.md [hunk 11]
```diff
@@ -268,7 +268,30 @@
...
+    if let Some(pop) = deque.pop_front() {
+    }
+
+    // 获取队列的长度
```
Comment (night-cruise) [inline docs/chapter_stack_and_queue/queue.md:290]: /* 获取队列的长度 */

File: docs/chapter_stack_and_queue/queue.md [hunk 12]
```diff
@@ -268,7 +268,30 @@
...
+    let size = deque.len();
+
+    // 判断队列是否为空
```
Comment (night-cruise) [inline docs/chapter_stack_and_queue/queue.md:293]: /* 判断队列是否为空 */

**Comment by hts0000** [comment]
我是要重新提交一次pull request吗？我第一次提交merge请求

**Review by night-cruise — APPROVED** [review]
Approve it.

**Review by krahets — APPROVED** [review]

**Pull request merged**